pkg/wallet/wallet.go

```diff
@@ -0,0 +1,93 @@
+package wallet
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"sync"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcutil/hdkeychain"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// Account represents a derived account.
+type Account struct {
+	path       accounts.DerivationPath
+	privateKey *ecdsa.PrivateKey
+}
+
+// Path returns the account's derivation path.
+func (a *Account) Path() accounts.DerivationPath {
+	return a.path
+}
+
+// Address returns the Ethereum address of the account.
+func (a *Account) Address() common.Address {
+	return crypto.PubkeyToAddress(a.privateKey.PublicKey)
+}
+
+// PrivateKey returns the account's private key.
+func (a *Account) PrivateKey() *ecdsa.PrivateKey {
+	return a.privateKey
+}
+
+// PublicKey returns the account's public key.
+func (a *Account) PublicKey() *ecdsa.PublicKey {
+	return &a.privateKey.PublicKey
+}
+
+// Wallet represents a master wallet that can generate accounts.
+type Wallet struct {
+	masterKey *hdkeychain.ExtendedKey
+	mx        sync.RWMutex
```

Comment:
I don't think we need it here, as we are not mutating the value of masterKey.
Is there something I miss @dimast-x ?

Reply:
Let's keep it in here to be safe